seed/seed.go

```diff
@@ -94,7 +103,51 @@ func (s *Seeder) Seed(cfg *config.Config) error {
 			log.Infof("consul value %s applied on field %s", *value, f.Name())
 			seedMap[f] = true
 		}
+		key, ok = ss[config.SourceFlag]
+		if ok {
+			var val string
+			flagset.StringVar(&val, key, "", "")
```

Comment:
Did you think of using the Lookup method of the flag package? If the flag exists get the returned flag and then the value and seed.
I did not try this but if this work the amount of code if low and very simple.
Could you try?

Reply:
I checked it: the `Lookup` method will always return a `Flag` if it is defined here with `StringVar`. The method won't tell us if the flag was set by the user or not and have to use `Visit` for this (see below in the code).